lib/Event/ScheduleCriteriaRequestEvent.php

```diff
@@ -0,0 +1,106 @@
+<?php
+/*
+ * Copyright (C) 2024 Xibo Signage Ltd
+ *
+ * Xibo - Digital Signage - https://xibosignage.com
+ *
+ * This file is part of Xibo.
+ *
+ * Xibo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Xibo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Xibo.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Xibo\Event;
+
+use Xibo\Support\Exception\ConfigurationException;
+
+class ScheduleCriteriaRequestEvent extends Event implements ScheduleCriteriaRequestInterface
+{
+    public static $NAME = 'schedule.criteria.request';
+    private $criteria = [];
+    private $currentTypeIndex = null;
+    private $currentMetric = null;
+
+    /**
+     * @inheritDoc
+     */
+    public function addType(string $id, string $type): self
+    {
+        // Initialize the type in the criteria array
+        $this->criteria['types'][] = [
+            'id' => $id,
+            'name' => $type,
+            'metrics' => []
+        ];
+
+        // Set the current type index for chaining
+        $this->currentTypeIndex = count($this->criteria['types']) - 1;
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addMetric(string $id, string $name): self
+    {
+        $metric = [
+            'id' => $id,
+            'name' => $name,
+            'values' => null
+        ];
+
+        // Add the metric to the current type
+        if (isset($this->criteria['types'][$this->currentTypeIndex])) {
+            $this->criteria['types'][$this->currentTypeIndex]['metrics'][] = $metric;
+            $this->currentMetric = $metric;
+        } else {
+            throw new ConfigurationException('Current type is not set.');
+        }
+
+        return $this;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addValues(string $inputType, array $values): self
+    {
+        // Restrict input types to 'dropdown', 'number', 'text' and 'date'
+        $allowedInputTypes = ['dropdown', 'number', 'text', 'date'];
+        if (!in_array($inputType, $allowedInputTypes)) {
+            throw new ConfigurationException('Invalid input type.');
```

Comment:
Although these are configuration errors on the server, we should still translate the message.